tests/BranchProtection2.py

```diff
@@ -0,0 +1,47 @@
+############################ Copyrights and license ############################
+#                                                                              #
+# Copyright 2018 Steve Kowalik <steven@wedontsleep.org>                        #
+#                                                                              #
+# This file is part of PyGithub.                                               #
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+from . import Framework
+
+
+class BranchProtection2(Framework.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.branch_protection = (
+            self.g
+            .get_repo("curvewise-forks/PyGithub")
+            .get_branch("master")
+            .get_protection()
+        )
+
+    def testAttributes(self):
```

Comment:
I'd prefer to add these three asserts to `BranchProtection.testAttributes` and record new replay data for `BranchProtection` with your fork branch.

Reply:
Hi @EnricoMi, I tried to do this in abb2bb9 , but was unable to capture new replay data. It failed to fetch PyGithub/integrations branch with 404, which I think is expected because this branch no longer exists.  Please see the full console output below. This is why I had originally moved the assertions out into a separate BranchProtections2 test, which ran against our fork. Is there a way to capture new replay data for tests run against the PyGithub/integrations branch? If not, could you please advise as to next steps?

Thank you for your help with this.

Console output: 

```
(base) jacob@pop-os:~/workspace/curvewise/PyGithub$ pytest -k BranchProtection --record
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.9.12, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/jacob/workspace/curvewise/PyGithub, configfile: pytest.ini
plugins: anyio-3.5.0, cov-4.0.0
collected 748 items / 747 deselected / 1 selected

tests/BranchProtection.py F                                                                                                                                                            [100%]

========================================================================================== FAILURES ==========================================================================================
______________________________________________________________________________ BranchProtection.testAttributes _______________________________________________________________________________

self = <tests.BranchProtection.BranchProtection testMethod=testAttributes>

    def setUp(self):
        super().setUp()
        self.branch_protection = (
>           self.g.get_user()
            .get_repo("PyGithub")
            .get_branch("integrations")
            .get_protection()
        )

tests/BranchProtection.py:30:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
github/AuthenticatedUser.py:1069: in get_repo
    headers, data = self._requester.requestJsonAndCheck(
github/Requester.py:402: in requestJsonAndCheck
    return self.__check(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <github.Requester.Requester object at 0x7f2b8ea62370>, status = 404
responseHeaders = {'DEBUG_FRAME': 1, 'access-control-allow-origin': '*', 'access-control-expose-headers': 'ETag, Link, Location, Retry-A...nterval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset', 'content-encoding': 'gzip', ...}
output = {'documentation_url': 'https://docs.github.com/rest/reference/repos#get-a-repository', 'message': 'Not Found'}

    def __check(self, status, responseHeaders, output):
        output = self.__structuredFromJson(output)
        if status >= 400:
>           raise self.__createException(status, responseHeaders, output)
E           github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-repository"}

github/Requester.py:427: UnknownObjectException
====================================================================================== warnings summary ======================================================================================
../../../anaconda3/lib/python3.9/site-packages/requests/__init__.py:102
  /home/jacob/anaconda3/lib/python3.9/site-packages/requests/__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (5.1.0)/charset_normalizer (2.0.4) doesn't match a supported version!
    warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== short test summary info ===================================================================================
FAILED tests/BranchProtection.py::BranchProtection::testAttributes - github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-repository"}
======================================================================== 1 failed, 747 deselected, 1 warning in 1.19s ========================================================================
```

Reply:
I think it is easiest to target a branch in your fork, which you can access and create as needed.

Reply:
Hi @EnricoMi, thanks for your reply. Should we try to maintain [these two assertions](
https://github.com/PyGithub/PyGithub/pull/2396/files#diff-261145a43f95fd792d69eafb577cd82b8f3a2d1fb090c5d756adf8b87b00ec26L37-L40
) from the original test case?

We don't currently have required status checks set up on any branch in our fork, but I could try to add them.

Let me know what you think.

Reply:
Hi @EnricoMi I just wanted to follow up on this. Thanks for your time and attention to this.

Reply:
Sorry for the delay. Yes, it would be great if you could add a `required_status_checks` to keep those two assertions.

Reply:
Completed in c9a9fec8c54ff2512d686b46b25c640b54506d14